### PR TITLE
Use Rust FFI #13: Multipart

### PR DESCRIPTION
**Pull request by tienvx**
Add multipart body support with example.

This PR depends on https://github.com/pact-foundation/pact-php/pull/332. Please review it first.
This PR depends on this [bug](https://github.com/pact-foundation/pact-reference/issues/314) to be fixed first.

### CODE REVIEW

File: example/multipart/consumer/tests/Service/HttpClientServiceTest.php [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            'profile_image' => $profileImageUrl,
+            'personal_note' => $personalNote,
+        ], \json_decode($userProfileResponse, true, 512, JSON_THROW_ON_ERROR));
+    }
+
+    private function createTempFile(string $contents): string
+    {
+        $path = tempnam(sys_get_temp_dir(), 'pact');
+        //$newPath = "$path.txt";
+        //rename($path, $newPath);
```
Comment (YOU54F) [inline example/multipart/consumer/tests/Service/HttpClientServiceTest.php:86]: still need these comments or can they be removed?

**Review by YOU54F — APPROVED** [review]
Looks good, couple of minor comments,

We should put a note about the content-type matching as well, and note the workaround in the test so the users can consider if they require the same for their own tests.

(I'm not sure of

**Comment by YOU54F** [comment]
happy to fix forward on the comments above ☝🏾 re documentation, in the upgrade guide

File: example/multipart/consumer/tests/Service/HttpClientServiceTest.php (hunk 1 above)
Comment (tienvx) [reply example/multipart/consumer/tests/Service/HttpClientServiceTest.php:86]: It's my bad to keep that comment there. It's better to mark this PR Draft. I will wait for 2 more fixes on pact-reference on multipart issues (one of them is https://github.com/pact-foundation/pact-reference/pull/319), then I will create new PR to delete this code completely (including this comment)

File: .cirrus.yml [hunk 2]
```diff
@@ -27,7 +27,7 @@ linux_arm64_task:
   container:
     image: php:$VERSION
   pre_req_script: 
-      - apt update --yes && apt install --yes zip unzip git libffi-dev
+      - apt update --yes && apt install --yes zip unzip git libffi-dev shared-mime-info
```
[the remark this answers is not in the document]
Comment (YOU54F) [reply .cirrus.yml:30]: ahhhh, it _should_ be an arm64 machine

this line should be `arm_container` for a host arm machine 🤦🏾 that is probably left over from using a local build, prior to cirrus merging the change to allow for `arm_container` to be resolved by the local cirrus-cli tool. 

https://github.com/tienvx/pact-php/blob/7fde6a44d6d6f7458786db19d42fa49199287323/.cirrus.yml#L27C1-L28C24

[the remark this answers is not in the document]
Comment (YOU54F) [reply .cirrus.yml:30]: Related issue

👉🏾 https://github.com/cirruslabs/cirrus-cli/pull/626#issuecomment-1568532398

File: example/multipart/consumer/tests/Service/HttpClientServiceTest.php (hunk 1 above)
Comment (tienvx) [reply example/multipart/consumer/tests/Service/HttpClientServiceTest.php:86]: Removed in https://github.com/pact-foundation/pact-php/pull/342

File: .cirrus.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (tienvx) [reply .cirrus.yml:30]: Update and revert in https://github.com/pact-foundation/pact-php/pull/344

**Pull request merged**